src/realm/table.cpp

```diff
@@ -1124,7 +1121,14 @@ void copy_list<String>(ref_type sub_table_ref, Obj& obj, ColKey col, Allocator&
         size_t list_size = from_list.size();
         auto l = obj.get_list<String>(col);
         for (size_t j = 0; j < list_size; j++) {
-            l.add(from_list.get_legacy(j));
+            auto val = from_list.get_legacy(j);
+            if (val.is_null() && !nullable) {
+                // Be forgiving about this
```

Comment:
Why should we be forgiving about this? Isn't it better to just throw a better exception that describe what is wrong?

Reply:
This error was also reported in Java here: https://github.com/realm/realm-java/issues/6995#issuecomment-675308734

It does feel slightly worrying if a list not allowed to contain Nulls do contain Nulls though? Any idea how this could happen?

Reply:
It is better to actually upgrade the file. If we just throw, there is nothing the user can do to fix the situation. I have not checked the core-5 code - yet.

Reply:
I guess I'm wondering how it is possible to come in this situation where the column is marked as not-null, but still contains null values? Doesn't that indicate another bug somewhere?

But agree that if there is nothing the user can do, then it is better to attempt to fix it instead of just leaving a broken Realm.

Reply:
Looked at it once more. Found another solution that actually returns the right value from list. Not sure if it is nicer.